consensus/istanbul/backend/val_enode_table.go

```diff
@@ -126,28 +139,29 @@ func (vet *validatorEnodeTable) pruneEntries(addressesToKeep map[common.Address]
 	for remoteAddress := range vet.valEnodeTable {
 		if !addressesToKeep[remoteAddress] {
 			log.Trace("Deleting entry from the valEnodeTable and reverseValEnodeTable table", "address", remoteAddress, "valEnodeEntry", vet.valEnodeTable[remoteAddress].String())
-			delete(vet.reverseValEnodeTable, vet.valEnodeTable[remoteAddress].enodeURL)
+			delete(vet.reverseValEnodeTable, vet.valEnodeTable[remoteAddress].node.ID())
 			delete(vet.valEnodeTable, remoteAddress)
 		}
 	}
 }
 
-func (vet *validatorEnodeTable) refreshValPeers(valSet istanbul.ValidatorSet, currentValPeers []string) {
+func (vet *validatorEnodeTable) refreshValPeers(valSet istanbul.ValidatorSet, currentValPeers map[enode.ID]consensus.Peer) {
 	vet.valEnodeTableMu.RLock()
 	defer vet.valEnodeTableMu.RUnlock()
 
 	// Add all of the valSet entries as validator peers
 	for _, val := range valSet.List() {
 		if valEnodeEntry := vet.getUsingAddress(val.Address()); valEnodeEntry != nil {
-			vet.addValidatorPeer(valEnodeEntry.enodeURL)
+			vet.p2pserver.AddPeerLabel(valEnodeEntry.node, "validator")
+			vet.p2pserver.AddTrustedPeerLabel(valEnodeEntry.node, "validator")
 		}
 	}
 
 	// Remove the peers that are not in the valset
-	for _, peerEnodeURL := range currentValPeers {
-		if peerAddress, ok := vet.reverseValEnodeTable[peerEnodeURL]; ok {
+	for id, valPeer := range currentValPeers {
+		if peerAddress, ok := vet.reverseValEnodeTable[id]; ok {
 			if _, src := valSet.GetByAddress(peerAddress); src == nil {
-				vet.removeValidatorPeer(peerEnodeURL)
+				vet.p2pserver.RemovePeerLabel(valPeer.Node(), "validator")
```

Comment:
Do you also need to call `RemoveTrustedPeerLabel`?

Reply:
Good catch!